spec/lib/user_comm_screener_spec.rb

```diff
@@ -162,4 +162,59 @@
       end
     end
   end
+
+  describe "actor preferences" do
+    fab!(:acting_user) { Fabricate(:user) }
+    fab!(:muted_user) { Fabricate(:muted_user, user: acting_user, muted_user: target_user1) }
+    fab!(:ignored_user) { Fabricate(:ignored_user, user: acting_user, ignored_user: target_user2, expiring_at: 2.days.from_now) }
+    fab!(:allowed_pm_user1) { AllowedPmUser.create!(user: acting_user, allowed_pm_user: target_user1) }
+    fab!(:allowed_pm_user2) { AllowedPmUser.create!(user: acting_user, allowed_pm_user: target_user2) }
+    fab!(:allowed_pm_user3) { AllowedPmUser.create!(user: acting_user, allowed_pm_user: target_user4) }
+
+    describe "#actor_preventing_communication" do
+      it "returns the user_ids of the users the actor is ignoring, muting, or disallowing PMs from" do
+        acting_user.user_option.update!(enable_allowed_pm_users: true)
+        expect(subject.actor_preventing_communication).to match_array([
+          target_user1.id, target_user2.id, target_user3.id, target_user5.id
+        ])
+      end
+    end
+
+    describe "#actor_allowing_communication" do
+      it "returns the user_ids of the users who the actor is not ignoring, muting, or disallowing PMs from" do
+        acting_user.user_option.update!(enable_allowed_pm_users: true)
+        expect(subject.actor_allowing_communication).to match_array([target_user4.id])
+      end
+    end
+
+    describe "#actor_ignoring?" do
+      it "returns true for user ids that the actor is ignoring" do
```

Comment:
This is related to [this comment](https://github.com/discourse/discourse/pull/17737/files#r935156555) because it makes describing the method tricky. Technically, this will only return true if the user is is part of `targer_users` which the class is initialized with. This will return false if I pass in a user id which the actor is ignoring but is not part of `target_users`.

Reply:
I suppose, but that is true of the existing methods as well -- for example if you call `ignoring_or_muting_actor?` with a user ID that is not in the original `target_users` it will give an incorrect result as well.

Maybe I should throw an error if a user ID that does not exist in `@target_users` is passed to any of the methods, that at least will prevent incorrect results from incorrect usage.

Reply:
Yup that sounds good :+1:

Reply:
Great, I think `Discourse::InvalidParameters` would be best for this, or maybe `Discourse::NotFound` actually since we cannot find the user_id passed in in the original target_users array